src/test/java/nextstep/subway/unit/LineTest.java

```diff
@@ -1,17 +1,150 @@
 package nextstep.subway.unit;
 
+import static org.assertj.core.api.Assertions.*;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.ArrayList;
+import java.util.List;
+import nextstep.subway.line.DownstreamStationIncludedException;
+import nextstep.subway.line.Line;
+import nextstep.subway.line.MismatchedUpstreamStationException;
+import nextstep.subway.line.NonDownstreamTerminusException;
+import nextstep.subway.line.Section;
+import nextstep.subway.line.SingleSectionRemovalException;
+import nextstep.subway.station.Station;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 
 class LineTest {
+
+    @DisplayName("세션을 성공적으로 추가한다")
+    @Test
+    void addSectionSuccess() {
+        // given
+        Line line = new Line();
+        Section section = new Section();
+
+        // when
+        line.addSection(section);
+
+        // then
+        assertThat(line.getSections()).contains(section);
+    }
+
+
+    @DisplayName("추가할 세션의 하행 스테이션이 Line에 포함 되여 있으면 예외를 던진다")
     @Test
-    void addSection() {
+    void addSectionFailedByUpStationOfNewSection() {
+        // given
+        Line line = new Line();
+        Section savedSection = new Section(1L,new Station("강남역"),new Station("양재역"),1);
+        line.addSection(savedSection);
+
+        // when,then
+        Section newSection = new Section(2L,new Station("양재역"),new Station("강남역"),1);
+        assertThatThrownBy(() -> line.addSection(newSection)).isInstanceOf(DownstreamStationIncludedException.class);
+    }
+
+    @DisplayName("노선 하행 종점역과 추가할 세션의 상행역이 일치하지 않아면 예외를 던진다")
+    @Test
+    void addSectionFailed() {
+        // given
+        Line line = new Line();
+        Section savedSection = new Section(1L,new Station("강남역"),new Station("양재역"),1);
+        line.addSection(savedSection);
+
+        // when,then
+        Section newSection = new Section(2L,new Station("교대역"),new Station("서초역"),1);
+        assertThatThrownBy(() -> line.addSection(newSection)).isInstanceOf(MismatchedUpstreamStationException.class);
     }
 
+    @DisplayName("노선에 포함한 스테이션을 가져온다")
     @Test
     void getStations() {
+        // given
+        Station gangnamStation = new Station("강남역");
+        Station yangjaeStation = new Station("양재역");
+        Line line = new Line();
+        Section section = new Section(1L, gangnamStation, yangjaeStation, 10);
+        line.addSection(section);
+
+        // when
+        List<Station> stations = line.getStations();
+
+        // then
+        assertThat(stations).contains(gangnamStation, yangjaeStation);
```

Comment:
```suggestion
        assertThat(stations).containsExactly(gangnamStation, yangjaeStation);
```
새로운 라인을 생성해서 구간을 추가한것이기 때문에 조금 더 정확한 검증을 해봐도 될 것 같아요 
지금은 새로운 구간을 더 추가해도 `then` 부분에 아무 영향을 미치지 않습니다 😅